File: projet full stack/Front-end/EspritEmbaucheApp/src/app/jobofferdto.ts
export class JobOfferDTO {
    
    idJobOffer: number;
    country: string;
    language: string;
    jobTitle: string;
    description: string;
    jobType: string;
    expLevel: string;
    offerDuration: string;

    companyId: number;
    companySname: string;
    nbrEmployees: string;
    address: string;
    numTel: string;
    fileName: string;

    dateSoumission: string; 
    daysSinceSubmission: number;

    hasExceededDuration: boolean;

    active: boolean;

  daysSinceExpiration: number;


  daysRemaining:number;

    startDate: Date;
    endDate: Date;
    offersduration: number;
    activejoboffer: boolean;

    websiteLink: string;


    constructor() {
      this.idJobOffer = 0;
      this.country = '';
      this.language = '';
      this.jobTitle = '';
      this.description = '';
      this.jobType = '';
      this.expLevel = '';
      this.offerDuration = '';
  
      this.companyId = 0;
      this.companySname = '';
      this.nbrEmployees = '';
      this.address = '';
      this.numTel = '';
      this.fileName = '';
  
      this.dateSoumission = '';
      this.daysSinceSubmission = 0;
      //this.visible = true;
    }

  }
  

File: projet full stack/back-end/EspritEmbauche/src/main/java/tn/esprit/Entities/JobOfferDTO.java
package tn.esprit.Entities;


import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Data
@RequiredArgsConstructor
@AllArgsConstructor
public class JobOfferDTO {
	
	private Long idJobOffer;
    private String country;
    private String language;
    private String jobTitle;
    private String description;
    private String jobType;
    private String expLevel;
    private String offerDuration;
    private LocalDate dateSoumission;
    private long daysSinceSubmission; 
    
    private LocalDate expirationDate;
    
    private boolean hasExceededDuration;
    
    private boolean active = true;
    
    
    private LocalDate startDate;
    private LocalDate endDate;
    private int daysRemaining;
    private int offersduration;    
    private boolean activejoboffer;
    
 
    private String websiteLink;
    
    private Long companyId;
    private String companySname;
    private String nbrEmployees;
    private String address;
    private String numTel;
    private String fileName;
    
    public boolean isActivejoboffer() {
        return activejoboffer;
    }

    public void setActivejoboffer(boolean activejoboffer) {
        this.activejoboffer = activejoboffer;
    }
    
    public Boolean getActivejoboffer() {
        return activejoboffer;
    }

     
    
}
